webapp/src/marked.tsx

```diff
@@ -299,13 +300,24 @@ export class MarkedContent extends data.Component<MarkedContentProps, MarkedCont
 
         pxt.Util.toArray(content.querySelectorAll('Video.ams-embed'))
             .forEach((inlineVideo: HTMLElement) => {
+
                 let player = MediaPlayer().create()
                 player.initialize(inlineVideo, inlineVideo.getAttribute("src"));
+                const src = inlineVideo.getAttribute('src');
+                let url = new URL(src);
+                const END_TIME = url.searchParams.get("endTime");
+                player.on(
+                    dashjs.MediaPlayer.events.PLAYBACK_TIME_UPDATED,
+                    (e: dashjs.PlaybackTimeUpdatedEvent) => {
+                        if (parseInt(END_TIME) <= e.time) {
+                            player.pause();
```

Comment:
One thing you might check for is a 'set time' function and set the current time in the video to END_TIME, so if the user tries to scrub past the end of the video section it gets 'locked' back to `END_TIME`. This is not strictly necessary though / doesn't have to be in this pr

Reply:
thanks for recommendation, i'll put this on to-do for later!